weather_mv/loader_pipeline/metrics.py

```diff
@@ -90,24 +89,35 @@ def wrapper(self, *args, **kwargs):
                 raise ValueError("Function is not a generator.")
 
         return wrapper
+
     return decorator
 
 
 class AddTimer(beam.DoFn):
-    """DoFn to add a empty time_dict per element in PCollection. This dict will stage_names as keys
-    and the time it took for that element in that stage."""
+    """DoFn to add a empty time_dict per element in PCollection. This dict will
+    stage_names as keys and the time it took for that element in that stage."""
+
     def process(self, element) -> t.Iterator[t.Any]:
-        time_dict = {}
+        time_dict = {
+            "uri": element,
+            "bucket": get_file_time(element),
+            "pickup": time.time(),
+        }
         yield element, time_dict
 
 
 class AddMetrics(beam.DoFn):
-    """DoFn to add Element Processing Time metric to beam. Expects PCollection to contain a time_dict."""
+    """DoFn to add Element Processing Time metric to beam. Expects PCollection
+    to contain a time_dict."""
 
-    def __init__(self, asset_start_time_format: str = '%Y-%m-%dT%H:%M:%SZ'):
+    def __init__(self, asset_start_time_format: str = "%Y-%m-%dT%H:%M:%SZ"):
         super().__init__()
-        self.element_processing_time = metric.Metrics.distribution('Time', 'element_processing_time_ms')
-        self.data_latency_time = metric.Metrics.distribution('Time', 'data_latency_time_ms')
+        self.element_processing_time = metric.Metrics.distribution(
+            "Time", "element_processing_time_ms"
+        )
+        self.data_latency_time = metric.Metrics.distribution(
+            "Time", "data_latency_time_ms"
+        )
         self.asset_start_time_format = asset_start_time_format
```

Comment:
let's rename this to asset_init_time.

Reply:
rename `data_latency_time`? We are not changing it. It is serving its purpose - data latency from asset_start_time to current_time.

Reply:
ahh, I'm talking about `asset_start_time`; my bad I should have mention it.